server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/create/CreateSnapshotRequest.java

```diff
@@ -379,29 +389,35 @@ public CreateSnapshotRequest featureStates(List<String> featureStates) {
     public CreateSnapshotRequest source(Map<String, Object> source) {
         for (Map.Entry<String, Object> entry : source.entrySet()) {
             String name = entry.getKey();
-            if (name.equals("indices")) {
-                if (entry.getValue() instanceof String) {
-                    indices(Strings.splitStringByCommaToArray((String) entry.getValue()));
-                } else if (entry.getValue() instanceof List) {
-                    indices((List<String>) entry.getValue());
-                } else {
-                    throw new IllegalArgumentException("malformed indices section, should be an array of strings");
-                }
-            } else if (name.equals("feature_states")) {
-                if (entry.getValue() instanceof List) {
-                    featureStates((List<String>) entry.getValue());
-                } else {
-                    throw new IllegalArgumentException("malformed feature_states section, should be an array of strings");
-                }
-            } else if (name.equals("partial")) {
-                partial(nodeBooleanValue(entry.getValue(), "partial"));
-            } else if (name.equals("include_global_state")) {
-                includeGlobalState = nodeBooleanValue(entry.getValue(), "include_global_state");
-            } else if (name.equals("metadata")) {
-                if (entry.getValue() != null && (entry.getValue() instanceof Map == false)) {
-                    throw new IllegalArgumentException("malformed metadata, should be an object");
-                }
-                userMetadata((Map<String, Object>) entry.getValue());
+            switch (name) {
```

Comment:
Admittedly sort of unrelated but it took me longer than it should have to track down how SLM configures its snapshots via this method and this change makes things at least a little easier to read IMO.